packages/relay/src/lib/eth.ts

```diff
@@ -730,6 +730,19 @@ export class EthImpl implements Eth {
 
       if (!gasPrice) {
         gasPrice = await this.getFeeWeibars(EthImpl.ethGasPrice, requestIdPrefix);
+
+        // converting to tinybar and afterward to weibar again is needed
+        // in order to handle the possibility of an invalid float number being calculated as a gas price
+        // e.g.
+        //   current gas price = 126
+        //   buffer = 10%
+        //   buffered gas price = 126 + 12.6 = 138.6 <--- invalid tinybars
+        gasPrice +=
```

Comment:
move to a util method outside of this class and add some tests to it

Reply:
Done.